pygfx/utils/show.py

```diff
@@ -16,9 +17,243 @@
+        if value is not None:
+            return value
+
+        # persist callback to prevent it from changing
+        self.__set__(instance, self.default_callback)
+        return self.default_callback
+
+    def __set__(self, instance, value) -> None:
+        setattr(instance, self.private_name, value)
+
+    def __call__(self, callback_fn) -> None:
+        self.default_callback = callback_fn
+
+
+class Display:
+    """A Helper to display an object or scene
+
+    This class provides the basic scaffolding needed to visualize a given
+    WorldObject. To do so the class chooses a sensible default for each part of
+    the full setup (@almarklein: what is the technical term for the full setup?)
+    unless the value is explicitly set by the user or exists as a child of
+    ``object``. For example, it will create a camera unless you explicitly set a
+    value for camera or if there is (at least) one camera in the scene.
+
+    Parameters
+    ----------
+    canvas : WgpuCanvas
+        The canvas used to display the object. If both ``renderer`` and
+        ``canvas`` are set, then the renderer needs to use the set canvas.
+    renderer : gfx.Renderer
+        The renderer to use while drawing the scene. If both ``renderer``
+        and ``canvas`` are set, then the renderer needs to use the set canvas.
+    controller : gfx.Controller
+        The camera controller to use. If not set, Display will use the first
+        controller defined on ``object``. If there is none, Display will
+        create one.
+    camera : gfx.Camera
+        The camera to use. If not set, Display will use the first camera
+        defined on ``object``. If there is none, Display will create one.
+    before_render : Callable
+        A callback that will be executed during each draw call before a new
+        render is made.
+    after_render : Callable
+        A callback that will be executed during each draw call after a new
+        render is made.
+    draw_function : Callable
+        Replaces the draw callback with a custom one. If set both
+        `before_render` and `after_render` will have no effect.
+
+    """
+
+    before_render = RenderCallback()
+    after_render = RenderCallback()
+    draw_function = RenderCallback()
+
+    def __init__(
+        self,
+        canvas=None,
+        renderer=None,
+        controller=None,
+        camera=None,
+        before_render=None,
+        after_render=None,
+        draw_function=None,
+    ) -> None:
+        self.canvas = canvas
+        self.renderer = renderer
+        self.controller = controller
+        self.camera = camera
+        self.draw_function = draw_function or self.default_draw
+        self.scene = None
+
+        if before_render is not None:
+            self.before_render = before_render
+
+        if after_render is not None:
+            self.after_render = after_render
+
+    def default_draw(self):
+        if self.before_render is not None:
```

Comment:
How can this ever be `None`? Doesn't it resolve to the class level RenderCallback always?